packages/ethereum-contracts/test/foundry/libs/SafeGasLibrary.prop.t.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity 0.8.19;
+
+import { SafeGasLibrary } from "@superfluid-finance/ethereum-contracts/contracts/libs/SafeGasLibrary.sol";
+
+import "forge-std/Test.sol";
+
+/// @title SafeGasLibraryProperties
+/// @author Superfluid
+/// @notice This contract tests that the SafeGasLibrary behaves as expected.
+/// @dev If the gasleft() inside the function is less than 1/63 of the gasleft() before the function call,
+/// then the function ran out of gas and we revert the whole transaction.
+/// We mock this here by using a multiple of the gasleft() before the function call.
+contract SafeGasLibraryProperties is Test {
+
+    function test_SafeGasRevertsAsExpected(uint16 multiple) public {
```

Comment:
This is a bad test, you are testing exactly how it is coded so you are not actually testing anything useful.

But what you really want to test is the property that: when you give an arbitrary gas limit to a call and if it reverts OUT_OF_GAS it is less than 63*.

Reply:
removed 😅